lib/_stream_readable.js

```diff
@@ -505,19 +505,10 @@ function onEofChunk(stream, state) {
   }
   state.ended = true;
 
-  if (state.sync) {
-    // If we are sync, wait until next tick to emit the data.
-    // Otherwise we risk emitting data in the flow()
-    // the readable code triggers during a read() call
-    emitReadable(stream);
-  } else {
-    // Emit 'readable' now to make sure it gets picked up.
-    state.needReadable = false;
```

Comment:
Is it ok to not set `needReadable` to `false`? Is this the reason why this could be semver-major?

Reply:
Good spot, fixed.